COLLABORATOR_GUIDE.md

```diff
@@ -0,0 +1,96 @@
+# jade Collaborator Guide
```

Comment:
I prefer having it in the repo because it increases its visibility. For example, when other devs evaluate using jade, they can see our process - and hopefully they like what they see ;)
We could always extract it into static site at a later point in time.
@rlidwka would you be ok with leaving it in a separate file for now?